cns/cnsclient/cnsclient_test.go

```diff
@@ -1,35 +1,71 @@
 package cnsclient
 
 import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
 	"net"
+	"net/http"
+	"os"
+	"reflect"
 	"strconv"
+	"testing"
 
 	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/common"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/google/uuid"
 )
 
 var (
-	testNCID = "06867cf3-332d-409d-8819-ed70d2c116b0"
+	svc *restserver.HTTPRestService
+)
 
-	testIP1      = "10.0.0.1"
-	testPod1GUID = "898fb8f1-f93e-4c96-9c31-6b89098949a3"
-	testPod1Info = cns.KubernetesPodInfo{
-		PodName:      "testpod1",
-		PodNamespace: "testpod1namespace",
-	}
+const (
+	primaryIp           = "10.0.0.5"
+	gatewayIp           = "10.0.0.1"
+	dockerContainerType = cns.Docker
 )
 
-// func addTestStateToRestServer(svc *restserver.HTTPRestService) {
-// 	// set state as already allocated
-// 	state1, _ := restserver.NewPodStateWithOrchestratorContext(testIP1, 24, testPod1GUID, testNCID, cns.Available, testPod1Info)
-// 	ipconfigs := map[string]cns.SecondaryIPConfig{
-// 		state1.ID: state1,
-// 	}
-// 	nc := cns.CreateNetworkContainerRequest{
-// 		SecondaryIPConfigs: ipconfigs,
-// 	}
+func addTestStateToRestServer(t *testing.T, secondaryIps []string) {
+	var ipConfig cns.IPConfiguration
+	ipConfig.DNSServers = []string{"8.8.8.8", "8.8.4.4"}
+	ipConfig.GatewayIPAddress = gatewayIp
+	var ipSubnet cns.IPSubnet
+	ipSubnet.IPAddress = primaryIp
+	ipSubnet.PrefixLength = 32
+	ipConfig.IPSubnet = ipSubnet
+	secondaryIPConfigs := make(map[string]cns.SecondaryIPConfig)
+
+	for _, secIpAddress := range secondaryIps {
+		secIpConfig := cns.SecondaryIPConfig{
+			IPSubnet: cns.IPSubnet{
+				IPAddress:    secIpAddress,
+				PrefixLength: 32,
+			},
+		}
+		ipId, err := uuid.NewUUID()
+		if err != nil {
+			t.Fatalf("Failed to generate UUID for secondaryipconfig, err:%s", err)
+		}
+		secondaryIPConfigs[ipId.String()] = secIpConfig
+	}
+
+	req := cns.CreateNetworkContainerRequest{
+		NetworkContainerType: dockerContainerType,
+		NetworkContainerid:   "testNcId1",
+		IPConfiguration:      ipConfig,
+		SecondaryIPConfigs:   secondaryIPConfigs,
+	}
 
-// 	svc.CreateOrUpdateNetworkContainerWithSecondaryIPConfigs(nc)
-// }
+	returnCode := svc.CreateOrUpdateNetworkContainerInternal(req)
+	if returnCode != 0 {
+		t.Fatalf("Failed to createNetworkContainerRequest, req: %+v, err: %d", req, returnCode)
```

Comment:
This test is failing right here when called from TestCNSClientRequestAndRelease. The svc.state.OrchestratorType is "kubernetes" for this test and fails on a check in CreateOrUpdateNetworkContainerInternal
![image](https://user-images.githubusercontent.com/35265851/88238361-4ec42880-cc36-11ea-98db-9e6d6829753f.png)
Run with the "-v" flag and you'll see
`/usr/local/go/bin/go test -v -timeout 30s -tags test github.com/Azure/azure-container-networking/cns/cnsclient -run ^TestCNSClientRequestAndRelease$`

Reply:
ah good catch, wonder why it passed in my local setup. fixed it